tempesta_fw/t/functional/helpers/control.py

```diff
@@ -363,20 +400,39 @@ def start(self):
         cmd = ' '.join([tf_cfg.cfg.get('Server', 'nginx'), '-c', config_file])
         self.node.run_cmd(cmd, ignore_stderr=True,
                           err_msg=(self.err_msg % ('start', self.get_name())))
+        self.state = "started"
+
+    def force_stop(self):
```

Comment:
both `Nginx` and `Tempesta` classes has the same "stateful" approach. (Seems quite the same for client classes.) Isn't it better to derive them from some common  class? 
As i see this, being statefull is very daemon specific: you can't wrap everything  in one big try-catch block, you have to divide the full procedure into small steps and have a separate try-catch block for every action. You've also noticed that and do it in the right way. We can make a base class for `stop()` functions that runs array of stop steps and define that array in every derived class.

Reply:
agree